proof_ipa.go

```diff
@@ -257,32 +300,45 @@ func SerializeProof(proof *Proof) (*VerkleProof, StateDiff, error) {
 			stemdiff = &statediff[len(statediff)-1]
 			copy(stemdiff.Stem[:], stem)
 		}
-		stemdiff.SuffixDiffs = append(stemdiff.SuffixDiffs, SuffixStateDiff{Suffix: key[StemSize]})
-		newsd := &stemdiff.SuffixDiffs[len(stemdiff.SuffixDiffs)-1]
-
-		var valueLen = len(proof.PreValues[i])
-		switch valueLen {
-		case 0:
-			// null value
-		case 32:
-			newsd.CurrentValue = (*[32]byte)(proof.PreValues[i])
-		default:
-			var aligned [32]byte
-			copy(aligned[:valueLen], proof.PreValues[i])
-			newsd.CurrentValue = (*[32]byte)(unsafe.Pointer(&aligned[0]))
-		}
-
-		valueLen = len(proof.PostValues[i])
-		switch valueLen {
-		case 0:
-			// null value
-		case 32:
-			newsd.NewValue = (*[32]byte)(proof.PostValues[i])
+		preLen := len(proof.PreValues[i])
+		postLen := len(proof.PostValues[i])
+		switch {
+		case isInsertion(preLen, postLen):
+			stemdiff.InsertedSuffixes = append(stemdiff.InsertedSuffixes, key[StemSize])
+			if postLen == 0 {
```

Comment:
Did you meant `!= 32`? 

If we're in an `isInsertion` then `postLen` can't be 0.